### DISCUSSION TITLE
文档 Target 似乎没法 listen 同样

### DISCUSSION THREAD

**Discussion by Mister-Hope** [YunYouJun/el-bot#18] [Q&A]
Target 按道理来说，支持的仅应为 `'admin'` `'master'` 还有 `{frirend?: number[]; group?: number[]}`

文档的描述令我很困惑。 target 是 friend 和group 没有道理，填一个数组更无从知道要发给群还是好友。

listen 填一个数组还可以理解，对于一个号码我就同时监听群和好友就行了。Target 我觉得不应该尝试同时以两种渠道发一遍

**Comment by Mister-Hope** [comment]
另外在 0.8.0 beta 7  版本， listen 和 target 的推导都是 any，没有做内置的类型。不知道新版本有没有加。

我的想法是我直接把我写好的东西按照官网推荐的格式提交给官方。所以这里我就特别的问一句，不然我满足自己实现就行了。

**Reply by YunYouJun** [reply]
`listen` 和 `target` 为了统一使用了同样的类型。
监听数组，是语法糖层面的兼容，为了 `listen: [123, 456]` 方便起见。但的确造成了语意不明确的问题，考虑统一只支持 `listen: {frirend?: (number | 'admin' | 'master')[]; group?: number[]}` 的形式。

或者是否其他更好的建议方案？

**Comment by Mister-Hope** [comment]
另外还有一个问题，插件函数的第二个参数是干嘛用的？ 是默认检测插件名称的 camelCase 键，自动传入么？文档里完全没有讲，但是 niubi 却用了。没仔细看代码。方便的时候把文档也补上？😉

**Reply by YunYouJun** [reply]
是的，第二个参数是根据插件名检测默认的配置项。（文档的确很多没写上，下次补。打算 0.8 正式版构建补充一遍，但是 mah 上了 2.0（还不稳定），打算放弃 0.8，直接 0.9 开发适配 2.0 了。）
以及设想的功能，因为时间精力有限，很多还没搞定，打算将其更加抽象一点，（譬如抽出 mirai 的适配器），去兼容一下一些通用的机器人框架，以避免一些重复功能的开发。
